src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -727,9 +728,18 @@ TEST_F(SingleNodeTest, TestDirectActorTaskSubmissionPerf) {
     RAY_CHECK_OK(
         driver.Tasks().SubmitActorTask(*actor_handle, func, args, options, &return_ids));
     ASSERT_EQ(return_ids.size(), 1);
+    object_ids.emplace_back(return_ids[0]);
   }
   RAY_LOG(INFO) << "finish submitting " << num_tasks << " tasks"
                 << ", which takes " << current_time_ms() - start_ms << " ms";
+
+  for (const auto &object_id : object_ids) {
+    std::vector<std::shared_ptr<RayObject>> results;
+    RAY_CHECK_OK(driver.Objects().Get({object_id}, -1, &results));
+    ASSERT_EQ(results.size(), 1);
+  }
+  RAY_LOG(INFO) << "finish executing " << num_tasks << " tasks"
+                << ", which takes " << current_time_ms() - start_ms << " ms";
```

Comment:
could you please also add a test to send a lot of tasks (say 100K) and verify they are received in the same order?

Reply:
IMO, we can leave this for an end-to-end integration test in Python (not sure how exactly to create python function descriptions from C++). The scheduling queue unit tests already verify it sequences calls correctly.

Reply:
Python worker is not yet migrated to core worker yet thus doing it in python is not possible now. We don't actually need the python function descriptors, and can just do it in C++ like existing core worker test cases, and in the receiving actor we can check if the requests are ordered.

While it's true that scheduling queue test covers the reordering, it would be good to have a integration test case with many requests to stress test it, since this is very critical code. What do you think?

Reply:
I agree we need an integration test. I don't know how to do it in C++ though, maybe you can provide an example?

Otherwise, we can leave a TODO to add it in Python (actually I believe there are already Python tests that cover this, which will automatically be applied once we switch to direct actor calls by default).

Reply:
Sure, actually a lot of the test cases in the `core_worker_test`.cc are integration tests. e.g. `TestActorTask`. The simplest way is to just copy the TestDirectActorTaskLocalPerformance in my original pr. The test case basically creates a driver, which creates an actor and submits a lot of tasks to that actor. There are some tweaks needed to basically pass a special argument to the actor creation task, and then in mock_worker you can check if this exists, and if it does then verify the data for the following tasks, assuming that the driver submits all the data from 1,2,3, ...

Reply:
Yeah, but how can you verify the ordering without being able to change the code in the actor?

Reply:
it is a bit hacky, as I mentioned above, you could modify the execution callback in mock_worker.cc to do this, which is called for each received task of the actor.